lib/new_playground.dart

```diff
@@ -609,13 +626,9 @@ class Playground implements GistContainer, GistController {
 
       _clearOutput();
 
-      if ((_layout == Layout.web) != gist.hasWebContent()) {
-        if (gist.hasWebContent()) {
-          _changeLayout(Layout.web);
-        } else {
-          // TODO (johnpryan): detect if app is a dart or flutter app
-          _changeLayout(Layout.dart);
-        }
+      if ((_layout == Layout.web) != gist.hasWebContent() ||
```

Comment:
Was this construction (`(a == b) != c`) used elsewhere in the code ? While I understand what it's doing, it's unusual enough that I needed a moment to parse it while reading through.

Is there enough of a cost to `_changeLayout` that you can't just call it with the appropriate layout every time, even if the layout is possibly in the correct configuration already?

Reply:
good point, fixed.